src/go/rpk/pkg/cli/cluster/partitions/move.go

```diff
@@ -0,0 +1,176 @@
+package partitions
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/adminapi"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/config"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/out"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+func newMovePartitionReplicasCommand(fs afero.Fs, p *config.Params) *cobra.Command {
+	var partitions []string
+	cmd := &cobra.Command{
+		Use:   "move",
+		Short: "Move partition replicas across nodes / cores",
+		Long:  helpAlterAssignments,
+		Run: func(cmd *cobra.Command, topics []string) {
+			p, err := p.LoadVirtualProfile(fs)
+			out.MaybeDie(err, "unable to load config: %v", err)
+			out.CheckExitCloudAdmin(p)
+
+			cl, err := adminapi.NewClient(fs, p)
+			out.MaybeDie(err, "unable to initialize admin client: %v", err)
+
+			for i, topic := range topics {
+				if !strings.Contains(topic, "/") {
+					topics[i] = "kafka/" + topic
+				}
+			}
+			for _, partition := range partitions {
+				if len(topics) > 0 { // foo -p 0:1,2,3
+					p, replicas, err := parsePartitionReplicas(partition)
+					out.MaybeDie(err, "unable to execute the request: %v\n", err)
+					for _, nt := range topics {
+						err = cl.MoveReplicas(cmd.Context(), nt+"/"+p, replicas)
+						out.MaybeDie(err, "unable to move partition replicas: %v\n", err)
+					}
+				} else { // -p foo/0:1,2,3
+					ntp, replicas, err := parseTopicPartitionReplicas(partition)
+					out.MaybeDie(err, "unable to execute the request: %v\n", err)
+					err = cl.MoveReplicas(cmd.Context(), ntp, replicas)
+					out.MaybeDie(err, "unable to move partition replicas: %v\n", err)
+				}
+			}
+			fmt.Print("Successfully submitted the request(s). Check the status with 'rpk cluster partitions move-status' or see the new assignments with 'rpk topic describe -p TOPIC'.\n")
+		},
+	}
+	cmd.Flags().StringArrayVarP(&partitions, "partition", "p", nil, "Partitions to move and new replica locations (repeatable)")
+	cmd.MarkFlagRequired("partition")
+	return cmd
+}
+
+// Expected input format is '0:1,2,3' or '1:1-0,2-0,3-0'
+// If '0:1,2,3' is passed, returns
+//
+// - '0'
+// - [{NodeID:1},{NodeID:2},{NodeID:3}]
+//
+// If '1:1-0,2-0,3-0' is passed, returns
+//
+// - '1'
+// - [{NodeID:1,Core:0},{NodeID:2,Core:0},{NodeID:3,Core:0}].
+func parsePartitionReplicas(partition string) (string, []adminapi.Replica, error) {
```

Comment:
If you want to merge both the following function and this one, and rely on a bit of cryptic regexp that will allow you to avoid error checking (since regexp will guarantee the format):
https://go.dev/play/p/pqUKAifdvV6

Reply:
Ha, it's cryptic and magical. I'll borrow the idea, thank you.

Reply:
Addressed in https://github.com/redpanda-data/redpanda/commit/1595272ed458ac47621cdbcad4c550c99140d622